docs/developer/best_practices/commit.rst

```diff
@@ -0,0 +1,356 @@
+Commit best practices
+---------------------
+
+How to split a change into commits
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The golden rule to create good commits is to ensure that there is only one
+logical change per commit.
```

Comment:
I'd rephrase this, because *logical* is quite broad and non-precise.

IMO, the content of a commit should be `a reversible unit`, i.e., something which, when `git revert`ed, doesn't break the product. Related, but less important (and likely more related to what you describe as `logical unit`) is something which can be `git cherry-pick`ed.

Reply:
Looks like you refer to this before. However, I still think *logical* is too imprecise.

Reply:
Yeah, `logical` is kinda unclear I agree.

I'll rewrite this part using a definition more similar to your yours.